### ISSUE TITLE
[MSAN] Does not detect uninitialized large (799+ bytes) structs across function calls

### ISSUE THREAD

**Issue by JustLinuxUser** [llvm/llvm-project#169108]
```c
// -Wall -Wextra -Wpedantic -fsanitize=memory -g -fno-inline-functions
#include <stddef.h>
#include <stdio.h>
#include <sys/types.h>

typedef struct s_large_struct
{
	char arr[800]; // With 799 the error is detected
	char other;
}	t_large_struct;

void	use_uninit(t_large_struct s)
{
	printf("el: %i\n", s.other);
}

int	main(void)
{
	t_large_struct	s;
	use_uninit(s);
}
```
[Godbolt](https://godbolt.org/z/317fW5czh)

For some reason msan stops tracking uninitialized structs across function calls when the function inlining is disabled, and the struct is larger then 799 bytes.
